api/v1beta1/metricstorage_webhook.go

```diff
@@ -55,8 +55,8 @@ func (spec *MetricStorageSpec) Default() {
 		// NOTE: If we want to enable dashboards in the future by default, set
 		//       it here like Alerting above
 	}
-	if spec.DataplaneNetwork == "" {
-		spec.DataplaneNetwork = "ctlplane"
+	if spec.DataplaneNetwork == nil || *spec.DataplaneNetwork == "" {
+		*spec.DataplaneNetwork = "ctlplane"
```

Comment:
```suggestion
		spec.DataplaneNetwork = ptr.To("ctlplane")
```
You'll need to import "k8s.io/utils/ptr" otherwise this will panic in case "spec.DataplaneNetwork == nil"

Reply:
After a discussion and trying a lot of different things, we found out kubernetes won't allow the field to be nil as there isn't the "nullable" property in the CRD. Kubernetes will always set the field to some value other than nil see: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable

So initializing the pointer isn't required at least until we decide we want to add the "nullable" property.